WMIAdventure/backend/WMIAdventure_backend/battle/businesslogic/card_states/BlockedCardState.py

```diff
@@ -0,0 +1,18 @@
+from typing import Any
+
+from battle.businesslogic.card_states.CardState import CardState
+
+
+class BlockedCardState(CardState):
+    """
+    Blocks card from being used by removing it's effects.
+    """
+
+    def card_used(self, effects_to_use: list[Any]):
+        """
+        Blocks card by clearing effects list.
+        :param effects_to_use: List of effects to be used in battle simulation.
+        :return: None.
+        """
+
+        effects_to_use.clear()
```

Comment:
Fajnie się ogląda tą architekturę i otwiera dużo możliwości, ale wygląda to jak zamknięcie konia w betonowym bunkrze zamiast w stajni, żeby nie uciekł. Czemu stan nie może być po prostu Enumem wykonującym te overridowane metody w switchu? Tam gdzieś dalej pojawia się taka dziwna rzecz, że karta może być w kilku stanach na raz i przez co w samej karcie jest więcej logiki niż w tych stanach.

Reply:
Powinna być możliwość nałożenia paru stanów na kartę, ale karta powinna być tylko w jednym stanie naraz - zrobię kolejkę stanów.

Reply:
Nie rozumiem z kolei czemu enum byłby lepszy